### ISSUE TITLE
OOM error for a specific batch size and image size when runing train_sup.py

### ISSUE THREAD

**Issue by Sara-Lei** [Haochen-Wang409/U2PL#110]
Hello, thank you for your amazing work. I came across a strange OOM error and I was hoping you could help.
I am trying to run train_sup.py on my customed dataset and came across an OOM error with image size = 320x320 and batch size = 4. I am using two GPUs.
However, when I increase or decrease the batch size or image size, the code can run without an OOM error. 
I did not use mixed precision.
When I use the same configuration and run train_semi.py with sup_only = 20, no OOM error either. 

(I trained using batch size=10 when running train_sup.py, no OOM. But when I ran train_semi.py with sup_only = 20, an OOM occurred.)

I have no idea what's happening here. Would really appreciate some help. Thank you!

**Comment by Haochen-Wang409** [comment]
Thanks for your attention to our work!
With batch_size=4, a batch of data contains 4 labeled images + 4 unlabeled images in ```train_semi.py```.
Therefore, the real batch_size should be 8.
Considering the extra contrastive head and other tensors, it is possible to encounter an OOM error with batch_size=8.

**Comment by Sara-Lei** [comment]
> Thanks for your attention to our work! With batch_size=4, a batch of data contains 4 labeled images + 4 unlabeled images in `train_semi.py`. Therefore, the real batch_size should be 8. Considering the extra contrastive head and other tensors, it is possible to encounter an OOM error with batch_size=8.

Sorry if I didn't make my statement clear. I simplified the questions and have sorted the situations in the table below:
![image](https://user-images.githubusercontent.com/65703260/224219913-294f8205-e64e-41bc-82e0-6bbe69325863.png)


When I tried to run train_sup.py, I only get OOM when using batch_size=4 and image_size=320. If I use a larger batch_size or image_size, I won't get OOM.

I was wondering if you know what might be the reason causing this? Thank you
